sources/Outputs/src/ParEvent.cpp

```diff
@@ -0,0 +1,93 @@
+//
+// Copyright (c) 2021, RTE (http://www.rte-france.com)
+// See AUTHORS.txt
+// All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, you can obtain one at http://mozilla.org/MPL/2.0/.
+// SPDX-License-Identifier: MPL-2.0
+//
+
+/**
+ * @file  ParEvent.cpp
+ *
+ * @brief Dynaflow launcher PAR file writer for events in a contigency implementation file
+ *
+ */
+
+#include "ParEvent.h"
+
+#include "Constants.h"
+#include "Contingencies.h"
+
+#include <DYNCommon.h>
+#include <PARParameter.h>
+#include <PARParameterFactory.h>
+#include <PARParametersSetCollection.h>
+#include <PARParametersSetCollectionFactory.h>
+#include <PARReference.h>
+#include <PARReferenceFactory.h>
+#include <PARXmlExporter.h>
+#include <boost/filesystem.hpp>
+
+namespace dfl {
+namespace outputs {
+
+namespace helper {
+
+template<class T>
+static boost::shared_ptr<parameters::Parameter>
+buildParameter(const std::string& name, const T& value) {
```

Comment:
Instead of creating a new file just containing the common `buildParameter` method, I choose to move this template to `Par.h`, and let `ParEvent.cpp` include `Par.h`.

Reply:
ParEvent and Par are different writers are shouldn't be related. it is better to use a different file

Reply:
I will create a `ParCommon.h`. It will only contain the definition of the `buildParameter` template method.

Reply:
The name would be more ParCommon.hpp. The naming rule for files is to use hpp extension if header contains template

Reply:
Done, I moved the template method common to Par and ParEvent writers to a separate file named `ParCommon.hpp`.

Reply:
The function should not be static in the header .hpp

Reply:
Also, why didn't you put it in Constants.h, which is the common header for outputs ?

Reply:
I thought the file `Constants.h` was reserved for constants (wouldn't be confusing to add other kind of declarations there?). Also, we are declaring a method that is only relevant for parameter writers, not for all outputs.

Reply:
OK

Reply:
Removed the static keyword from function template.